### PR TITLE
Move AwsS3V4Signer to auth

### PR DESCRIPTION
**Pull request by zoewangg**
- Move AwsS3V4Signer and related classes to auth
- Mark protected api in auth to internal api

### CODE REVIEW

File: auth/src/main/java/software/amazon/awssdk/auth/signer/internal/AwsSignerExecutionAttribute.java [hunk 1]
```diff
@@ -59,6 +59,16 @@
      */
     public static final ExecutionAttribute<Instant> PRESIGNER_EXPIRATION = new ExecutionAttribute<>("PresignerExpiration");
 
+    /**
+     * The key to specify whether to enable chunked encoding or not
+     */
+    public static final ExecutionAttribute<Boolean> ENABLE_CHUNKED_ENCODING = new ExecutionAttribute<>("ChunkedEncoding");
```
Comment (varunnvs92) [inline auth/src/main/java/software/amazon/awssdk/auth/signer/internal/AwsSignerExecutionAttribute.java]: I don't think these params are applicable to AwsSignerExecutionAttribute class. As of now, they are only applicable for S3 signer. 
Maybe we can  move them to a separate class or add ReviewBeforeRelease tag to finalize before GA.

Comment (zoewangg) [reply auth/src/main/java/software/amazon/awssdk/auth/signer/internal/AwsSignerExecutionAttribute.java]: yeah, I'll move them to a separate class

**Comment by zoewangg** [comment]
@varunnvs92 hmm, moving the tests to auth might be a bit tricky since we need to do extra stuff like creating bucket object and deleting afterwards as well.

**Comment by varunnvs92** [comment]
Thats true, we might need a bit more work. Anyways It is not a blocker for this change.

**Pull request merged**